unison-core/src/Unison/HashQualified.hs

```diff
@@ -51,9 +52,16 @@ toName = \case
 --   [.foo.bar, foo.bar] -> [foo.bar, .foo.bar]
 sortByLength :: [HashQualified] -> [HashQualified]
 sortByLength hs = sortOn f hs where
-  f (NameOnly n) = (Name.countDots n, 0, Left n)
-  f (HashQualified n _h) = (Name.countDots n, 1, Left n)
-  f (HashOnly h) = (maxBound, 0, Right h)
+  -- comparator: (num segments, is hash qualified, is absolute, name itself)
+  f :: HashQualified -> Either (Int, Bool, Bool, [NameSegment]) ShortHash
```

Comment:
Reworked this comparator but its behavior should be roughly the same. The only thing that I intended to change was more accurately ordering absolute/relative.

Before: an absolute name's leading dot just counted against it as if it was a name segment.
After: a relative name is preferred to an absolute name

Reply:
Seems reasonable.